services/src/test/scala/cromwell/pipeline/service/ProjectServiceTestImpl.scala

```diff
@@ -0,0 +1,87 @@
+package cromwell.pipeline.service
+import cromwell.pipeline.datastorage.dao.repository.ProjectRepositoryTestImpl
+import cromwell.pipeline.datastorage.dto._
+import cromwell.pipeline.model.wrapper.UserId
+import cromwell.pipeline.service.ProjectService.Exceptions.{ ProjectAccessDeniedException, ProjectNotFoundException }
+import cromwell.pipeline.service.VersioningException.RepositoryException
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+class ProjectServiceTestImpl(testMode: ServiceTestMode, projectRepository: ProjectRepositoryTestImpl)(
```

Comment:
I have a few questions:
1. Can we use trait ProjectRepository instead of ProjectRepositoryTestImpl?
2.  Can you put testMode value to the implicit and carry place ?
```suggestion
class ProjectServiceTestImpl(projectRepository: ProjectRepository)(implicit testMode: ServiceTestMode)(
```

Reply:
Nice idea, fixed